### PR TITLE
Instrument code to send Analytics hits

### PR DESCRIPTION
**Pull request by chanseokoh**
Part of #140.

Tracking client-side, opt-in usage on launching the new project wizard and completing the wizard.

### CODE REVIEW

**Comment by chanseokoh** [comment]
FYI, I checked that these event pings show up on GA.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java [hunk 1]
```diff
@@ -62,6 +64,9 @@ public MavenAppEngineStandardWizardPage() {
 
   @Override
   public void createControl(Composite parent) {
+    AnalyticsPingManager.getInstance().sendPing(
+        AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD, null, null, parent.getShell());
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java]: I suspect we should have separate events for the regular new project wizard and the maven new project wizard, or some sort of parameter that enables us to distinguish them

File: plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsEvents.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+public class AnalyticsEvents {
+
+  // Google Analytics event actions
+  public static final String LOGIN_START = "user.login.start";
+  public static final String LOGIN_COMPLETE = "user.login.complete";
+
+  public static final String APP_ENGINE_DEPLOY = "appengine.deploy";
+  public static final String APP_ENGINE_DEPLOY_SUCCESS = "appengine.deploy.success";
+  public static final String APP_ENGINE_LOCAL_DEV_RUN = "appengine.local.dev.run";
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsEvents.java]: LOCAL_SERVER

**Comment by chanseokoh** [comment]
Now distinguishing maven-based and non-maven project creation with extra metadata (`type=maven` vs. `type=native`).

File: plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsEvents.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  public static final String APP_ENGINE_NEW_PROJECT_WIZARD = "appengine.new.project.wizard";
+  public static final String APP_ENGINE_NEW_PROJECT_WIZARD_COMPLETE =
+      "appengine.new.project.wizard.complete";
+
+  // Metadata keys
+  public static final String APP_ENGINE_NEW_PROJECT_WIZARD_TYPE = "type";
+
+  // Metatata values
+  public static final String APP_ENGINE_NEW_PROJECT_WIZARD_TYPE_NATIVE = "native";
+  public static final String APP_ENGINE_NEW_PROJECT_WIZARD_TYPE_MAVEN = "maven-based";
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsEvents.java]: maybe just "maven"

**Comment by elharo** [comment]
LGTM

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java [hunk 4]
```diff
@@ -62,6 +64,11 @@ public MavenAppEngineStandardWizardPage() {
 
   @Override
   public void createControl(Composite parent) {
+    AnalyticsPingManager.getInstance().sendPing(
+        AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD,
+        AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE,
+        AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE_MAVEN, parent.getShell());
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java]: Would it be useful to log the archetype used?

Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java]: +1 for archetype

Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java]: Ah, I missed this. I'll file bugs for this.

**Pull request merged**